crates/project/src/lsp_store.rs

```diff
@@ -1405,11 +1406,13 @@ impl LspStore {
                 let result = lsp_request.await;
 
                 let response = result.map_err(|err| {
-                    log::warn!(
-                        "Generic lsp request to {} failed: {}",
-                        language_server.name(),
-                        err
-                    );
+                    cx.update(|cx| {
+                        this.update(cx, |_, cx| {
+                            cx.emit(LspStoreEvent::Notification(err.to_string()));
+                        })
+                    })
+                    .log_err();
```

Comment:
The repercussions of this change is that users are alerted with more lsp errors than they're used to in Zed. I'm unsure of the entire breadth of errors that may fall into this generic category that may be skipped in vscode.

Reply:
I think this is quite disrupting given that notifications aren't closing on their own and it's not possible to use the keyboard to close these.

At worst, we need to make this configurable and disabled by default.
What seems better, is to use the activity indicator, similar how formatter errors are doing: https://github.com/zed-industries/zed/blob/f6850cebf5a4263e7370db592102d0e4cec491d7/crates/activity_indicator/src/activity_indicator.rs#L485

That would require more tinkering, though, as messages seem to replace one another by default, and we want to show multiple at once.

Reply:
Thanks. I'll try out the activity indicator. 

Only because I'm unaware of what the indicator looks like: It seems more viable that we should be able to press escape key to remove notifications when they're visible, while also respecting Vim mode and any other subsystem that handles the escape key. Unsure of the work it'd take to do this though.

Reply:
Thinking through possible avenues

1. Have multiple activity indicator messages side by side or somehow stacked on another. Perhaps only two max, and truncate their lengths. But the presence of a diagnostic error along side the activity indicator is also tricky. I feel even having 2 activity indicators take too much horizontal space. 
<img width="562" alt="Screenshot 2024-12-11 at 3 53 56 AM" src="https://github.com/user-attachments/assets/3684f1cd-9b72-409e-8d02-542299d2bfa4">

So I guess that leaves us with showing multiple N last messages on hover of the activity status. We'll show up to the latest 2, each on their own line. The UI that appears on hover doesn't seem to wrap the message, so we'll let them trail on. If users want to see the full logs for either of them, they'd have to click to view the entire zed log.

2. Add new keybinding (escape key) to hide toasts. VSCode handles toast hiding globally from what I can tell.
```json
{ 
   "key": "escape",                
   "command": "notifications.hideToasts",
   "when": "notificationToastsVisible" 
}
```

Reply:
I much prefer the escape + toast option though. Using the toast's auto-hide functionality for LSP errors seems fair too.

Reply:
1.
I think we can add some `(N more)` prefix/suffix to that bar and actually show more items on click (not on hover), if we want to follow that path.

For 2. , I think we still need some per-language-server setting which we can press: if any server starts spamming with these messages, dismissing them with the keyboard is, of course, much better than with the mouse, but still somewhat distracting.
Presumably, we can add some button on that toast, that will change that toggle.

--------------

Overall, I think the toggle is good too, just not as safe as the other approach.
But both have their caveats.